usr/src/lib/libnsl/rpc/svc_generic.c

```diff
@@ -106,6 +107,221 @@ __svc_free_xprtlist(void)
 	__svc_free_xlist(&_svc_xprtlist, &xprtlist_lock);
 }
 
+static void
+error(char *errmsg, int sv_errno)
+{
+	if (t_errno != TSYSERR) {
+		(void) syslog(LOG_ERR, "%s: %s\n", errmsg, t_strerror(t_errno));
+	} else {
+		(void) syslog(LOG_ERR, "%s: %s\n", errmsg, strerror(sv_errno));
+	}
+}
+
+/*
+ * This function accomplishes the same thing as svc_create, except that it
+ * allows a port to be specified.  If no port is specified, svc_create is
+ * called.  Otherwise, we will loop through every network device in the given
+ * nettype, and register the service on that device, on the provided port.
+ */
+int
+svc_create_port(void (*dispatch)(), const rpcprog_t prognum,
+    const rpcvers_t versnum, const char *nettype, in_port_t port)
```

Comment:
It's worth specifying in the comment that the expectation is that the caller passes port in network byte order.